Data formatter

pclee777 opened the pull request:
Minimally functional data formatter. Both the PSAS Telemetry viewer and the PSAS packet serializers need to be installed to receive packets from the data formatter - installation directions found in their respective repositories.

https://github.com/psas/telemetry
https://github.com/psas/psas_packet

Commit:
Minimally functional data formatter

Added a file provided by Cort to be used in test_mode - gpio.rs.

Commit:
Whitespace edit - Tabs to spaces.

Commit:
Design document changes. Minor change to data_fmt::send_packet.

cwalexand commented:
How was this tested? What were the results?

---

Flight mode won't compile:

../common/data_fmt.rs:12:1: 12:24 error: can't find crate for `byteorder` [E0463]
../common/data_fmt.rs:12 extern crate byteorder;

---

data_fmt.rs:

Anything called after initialization probably shouldn't use unwrap() because that will panic during flight. There are 11 cases of unwrap() in this file -- should we do something different? Try to recover or ignore it?

The send_packet function never returns an error. Even if it did, main would just print out a number that wouldn't help the user debug much.

The flush_telemetry function's header says it has return values, but it doesn't.

The as_message function doesn't need to be pub.

---

gpio.rs (test mode):

Nice for this branch, but probably don't want to merge it to master.

Commit:
Added byteorder dependency for flight mode, removed unwrap, and fixed data_fmt::send_packet; telemetry viewer should receive packets again.

pclee777 commented:
- For testing: In one terminal window, the telemetry viewer was run, and then a browser was opened using localhost:8080 in the address bar. In another terminal window, flight/test_mode was ran afterwards. The telemetry viewer will display the receiving of packets - reduce data_fmt::P_LIMIT to 26 if you don't want to wait for the buffer to fill.
- Proper error handling will be implemented later.
- gpio.rs was added to resolve an import error for control module's for MyPins and Direction.

Commit:
Removed unnecessary "mut".

Commit:
Reimplemented Error handling.

Replaced all instances of “unwrap” with “try”.

Fixed issue where only the very first packet would yield “Reader
Broke!” by the telemetry reader.

Commit:
Merge branch 'master' into data_formatter

# Conflicts:
#	documents/Design.md
#	src/common/main.rs
#	src/flight_mode/Cargo.toml
#	src/test_mode/src/gpio.rs

Commit:
Copied master's gpio.rs over data_formatter branch gpio.rs

Commit:
Updated Design.md document
